core/src/test/scala/integration/kafka/api/BaseConsumerTest.scala

```diff
@@ -34,13 +35,15 @@ import scala.collection.Seq
  */
 abstract class BaseConsumerTest extends AbstractConsumerTest {
 
-  @Test
-  def testSimpleConsumption(): Unit = {
+  @ParameterizedTest
+  @ValueSource(strings = Array("generic", "consumer"))
```

Comment:
Is it possible to have a different type of 'source' that can be defined once vs. on each test? More of a nit, but curious.

Reply:
Perhaps, but I want to be able to have a different array for each test to enable them to be turned on individually.

Reply:
This has been replaced with a `MethodSource` that is capable of returning whatever combination we want.